addons/website_crm/models/website_visitor.py

```diff
@@ -43,6 +44,20 @@ def _check_for_message_composer(self):
             return True
         return check
 
+    def _inactive_visitors_domain(self):
+        """ Visitors tied to leads are considered always active and should not be deleted. """
+        domain = super(WebsiteVisitor, self)._inactive_visitors_domain()
+        return expression.AND([domain, [('lead_ids', '=', False)]])
```

Comment:
Fearing this might blow up on an heavy DB... Aka: shoudln't we build an sql query instead ? Not sure, asking for a friend.

Reply:
I'm guessing it will search within the many2many table between crm.lead and website.visitor, which I don't believe is super big?
Unless we create a crazy amount of leads through the contact form.

Reply:
It will return a domain [('id', 'in', ALL VISITORS WITHOUT LEADS)], aka potentially all visitors. We might want either a has_lead field, either using sub-queries instead of a domain leaf that will be evaluated and replaced.